crates/bevy_math/src/curves/variable_linear.rs

```diff
@@ -0,0 +1,243 @@
+use crate::{
+    curves::{Curve, CurveCursor, CurveError},
+    interpolation::Lerp,
+};
+
+// TODO: Curve/Clip need a validation during deserialization because they are
+// structured as SOA (struct of arrays), so the vec's length must match
+
+// TODO: impl Serialize, Deserialize
+/// Curve with sparse keyframes frames, in another words a curve with variable frame rate;
+///
+/// This is a very useful curve, because it can accommodate the output of a linear reduction keyframe algorithm
+/// to lower the memory foot print. As a down side it requires the use of a keyframe cursor, and
+/// loses performance when the curve frame rate is higher than the curve sampling frame rate;
+///
+/// It can't handle discontinuities, as in two keyframes with the same timestamp.
+///
+/// **NOTE** Keyframes count is limited by the [`CurveCursor`] size.
+#[derive(Default, Debug)]
+pub struct CurveVariableLinear<T> {
+    time_stamps: Vec<f32>,
+    keyframes: Vec<T>,
```

Comment:
Is there a specific reason to not use `Vec<(f32, T)>`?

Reply:
Yes, TLDR: Is usually faster.

The `sample` function perform a search to find the right keyframes for any given sample time, this search can be made more efficient by not pouting the cache with unused values like T in this case. This translates to faster random time sampling, but it also makes continuous sampling a bit faster